src/base-input.js

```diff
@@ -171,7 +171,13 @@ export default class BaseInput extends LitElement {
    * @returns {string} Validates string entered into the input field.
    */
   handleInput() {
+    // Prevent non-numeric characters from being entered on credit card fields
+    if (this.type === 'credit-card') {
+      this.inputElement.value = this.inputElement.value.replace(/[\D]/g,'');
```

Comment:
Will this remove the spaces that Cleave adds as well? Is that a problem?

Reply:
This is a little bit of a mixed bag:

If we allow the user to type spaces then they can enter something like "4444 44              432" which is undesirable. 

By only allowing the user to type only numbers then we do run into a situation where the following occurs:

1. Enter a value of "4444 4444 4444" (incomplete number
2. Change that value to "4444 444K 4444"

The formatting system will auto change it to "44444444444" (removing the K and all spaces). The next numeric character entered will reformat the value to a proper credit card format.

I chose scenario 2 as the lesser of two evils.